contest/register_contest.py

```diff
@@ -41,3 +49,82 @@ def register_group(group, contest):
         return
     for member in group.member.all():
         register_user(member, contest)
+
+def register_anonymous(contest, account_num):
+    if not is_integer(account_num):
+        logger.warning('Contest: input word is not interger! Can not register anonymous!')
+        return False
+    account_num = int(account_num)  
+    if account_num < 0:
+        logger.warning('Contest: input word is less than 0. Can not register anonymous!')
+        return False
+    if account_num > settings.MAX_ANONYMOUS:
+        too_many_anonymous_warning = 'Contest: register anonymous more than ' \
+             + str(settings.MAX_ANONYMOUS) + '! Set to ' + str(settings.MAX_ANONYMOUS) + '!'
+        logger.warning(too_many_anonymous_warning)
+        account_num = settings.MAX_ANONYMOUS
+
+    public_users = get_public_users()
+    available = 0
+    available_anonymous = []
+    for user in public_users:
+        if not attends_not_ended_contest(user):
+            user.is_active = False
+            available_anonymous.append(user)
+            available += 1
+
+    #if attend more than request, then kick some out
+    all_contestant = User.objects.filter(user__contest = contest).order_by('-user')
+    attended_anonymous = []
+    for user in all_contestant:
+        if user.username.startswith(settings.ANONYMOUS_PREFIX):
+            attended_anonymous.append(user)
+
+    attended = len(attended_anonymous)
+    if account_num == attended:
+        return True
+
+    if (account_num < attended):
+        for index,user in enumerate(attended_anonymous):
+            if index == (attended-account_num):
+                return True
+            username = user.username
+            contestant = Contestant.objects.get(user = user,contest = contest)
+            contestant.delete()
+            logger.info('Contest: User %s leave Contest %s!' % (username, contest.id))
+            
+    still_need = account_num - attended
+    #available is more than request
+    if(available >= still_need):
+        for user in available_anonymous:
+            if (still_need > 0):
+                if can_register(user, contest):
```

Comment:
An anonymous user will not login, so it can not pass ```can_register``` checking.
Maybe we don't have to check can_register for anonymous.